avocado/utils/genio.py

```diff
@@ -174,3 +174,16 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+class GenIOException(Exception):
+    pass
+
+
+def write_file_or_log(filename, data):
```

Comment:
Actually in this version it's function is not `write_file_or_log` but `write_file_or_fail` as you changed it to `raise Exception`.